arch/x86/crc32_fold_pclmulqdq.c

```diff
@@ -380,6 +387,131 @@ Z_INTERNAL void crc32_fold_copy_pclmulqdq(crc32_fold *crc, uint8_t *dst, const u
     crc32_fold_save_partial((__m128i *)crc->fold, xmm_crc_part);
 }
 
+#define ONCE(op) if (first) { \
+    first = 0; \
+    (op); \
+}
+#define XOR_INITIAL(where) ONCE(where = _mm_xor_si128(where, xmm_initial))
+
+Z_INTERNAL void crc32_fold_pclmulqdq(crc32_fold *crc, const uint8_t *src, size_t len, uint32_t init_crc) {
+    unsigned long algn_diff;
+    __m128i xmm_t0, xmm_t1, xmm_t2, xmm_t3;
+    __m128i xmm_crc0, xmm_crc1, xmm_crc2, xmm_crc3, xmm_crc_part;
+    __m128i xmm_initial = _mm_cvtsi32_si128(init_crc);
+    int32_t first = 1;
+
+    assert(len >= 64);
```

Comment:
Is this `assert` correct? I think this function could handle `len < 64`.

Reply:
I think it'd need to be at least 32, though technically we never call it for input sizes under 64 bytes.  The reason being is for how we handle the aligning partial load.  Instead of xor'ing the input bytes on the partial fold (where we don't yet have enough input), we do it on the the next set of bytes that will have at least 4 to accommodate the 4 byte initial CRC.

Reply:
Adjusted assertion statement and added explanatory comment in case someone gets too bold with it.